client/tasks/task-list-menu.tsx

```diff
@@ -14,13 +15,22 @@ export type TaskListMenuProps = {
 export const TaskListMenu: React.FC< TaskListMenuProps > = ( { id } ) => {
 	const { hideTaskList } = useDispatch( ONBOARDING_STORE_NAME );
 
+	const { invalidateResolutionForStoreSelector } = useDispatch(
+		OPTIONS_STORE_NAME
+	);
+
+	const onHideTaskList = useCallback( () => {
+		hideTaskList( id );
+		invalidateResolutionForStoreSelector( 'getOption' );
```

Comment:
What's the purpose behind invalidating `getOption`?  Seems like it would invalidate a lot of resolvers outside of this component as well.

Reply:
Oh, I think this might be so that other items (e.g., Activity Panels) get updated when this is changed, right?

Maybe we could `getTaskLists` in those items instead and check if the task list is hidden.

Reply:
I like this idea. Managed to make it happen in [`5876e08` (#7733)](https://github.com/woocommerce/woocommerce-admin/pull/7733/commits/5876e0835c1deb2c95bb2981648a3e19b121129e), with no invalidation necessary.